be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MeetingValidator.scala

```diff
@@ -70,24 +86,52 @@ sealed class MeetingValidator(dbActorRef: => AskableActorRef) extends MessageDat
 
-        if (!validateTimestampStaleness(data.creation)) {
-          Right(validationError(s"stale 'creation' timestamp (${data.creation})"))
-        } else if (!validateTimestampOrder(data.creation, data.last_modified)) {
-          Right(validationError(s"'last_modified' (${data.last_modified}) timestamp is smaller than 'creation' (${data.creation})"))
-        } else if (!validateTimestampStaleness(data.start)) {
-          Right(validationError(s"stale 'start' timestamp (${data.start})"))
-        } else if (data.end.isDefined && !validateTimestampOrder(data.creation, data.end.get)) {
-          Right(validationError(s"'end' (${data.end.get}) timestamp is smaller than 'creation' (${data.creation})"))
-        } else if (data.end.isDefined && !validateTimestampOrder(data.start, data.end.get)) {
-          Right(validationError(s"'end' (${data.end.get}) timestamp is smaller than 'start' (${data.start})"))
-        } else if (!validateWitnessSignatures(data.modification_signatures, data.modification_id)) {
-          Right(validationError("witness key-signature pairs are not valid for the given modification_id"))
-        } else if (expectedHash != data.id) {
-          Right(validationError("unexpected id"))
-        } else {
-          Left(rpcMessage)
-        }
+        runChecks(
+          checkTimestampStaleness(
+            rpcMessage,
+            data.creation,
+            validationError(s"stale 'creation' timestamp (${data.creation})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.creation,
+            data.last_modified,
+            validationError(s"'last_modified' (${data.last_modified}) timestamp is smaller than 'creation' (${data.creation})")
+          ),
+          checkTimestampStaleness(
+            rpcMessage,
+            data.start,
+            validationError(s"stale 'start' timestamp (${data.start})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.creation,
+            data.end.get,
+            validationError(s"'end' (${data.end.get}) timestamp is smaller than 'creation' (${data.creation})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.start,
+            data.end.get,
+            validationError(s"'end' (${data.end.get}) timestamp is smaller than 'start' (${data.start})")
+          ),
+          checkId(rpcMessage, expectedHash, data.id,validationError(s"unexpected id")),
+          checkWitnessSignatures(
+            rpcMessage,
+            data.modification_signatures,
+            data.modification_id,
+            validationError("witness key-signature pairs are not valid for the given modification_id")
+          )
+        )
       case _ => Right(validationErrorNoMessage(rpcMessage.id))
     }
   }
+
+  private def checkWitnessSignatures(rpcMessage: JsonRpcRequest, witnessesKeyPairs: List[WitnessSignaturePair], data: Hash, error: PipelineError): GraphMessage = {
```

Comment:
This overlaps with the PR made by Kilian on the Lao refactoring. Same comment as him, this function could be moved in the same place as `validateWitnessSignatures`. 
Also, when defining new functions, do not forget to comment them for more clarity :)